sftp_blueprints/move_file.py

```
@@ -0,0 +1,216 @@
+
+
+def find_sftp_file_names(client, prefix=''):
+    """
+    Fetched all the files in the folder on the SFTP server
+    """
+    try:
+        files = []
+        folders = []
+        if prefix != '':
+            data = client.listdir(prefix)
+        else:
+            data = client.listdir()
+
+        for fname in data:
+            if fname.startswith('.'):
+                continue
+            if prefix:
+                fdata = str(client.lstat(f'{prefix}/{fname}')).split()[0]
+            else:
+                fdata = str(client.lstat(fname)).split()[0]
+            if fdata.startswith('d'):
+                folders.append(fname)
+            else:
+                if prefix != '':
+                    files.append(f'{prefix}/{fname}')
+                else:
+                    files.append(fname)
+
+        for folder in folders:
+            if prefix:
+                folder = f'{prefix}/{folder}'
+            files.extend(find_sftp_file_names(client, folder))
+    except Exception as e:
+        print(f'Failed to find files in folder {prefix}')
+        raise(e)
+
+    return files
+
+
+def cd_into_cwd(client, destination_path):
+    """
+    Changes working directory to the specified destination path
+    and creates it if it doesn't exist
+    """
+    for folder in destination_path.split('/'):
+        try:
+            client.chdir(folder)
+        except Exception:
+            client.mkdir(folder)
+            client.chdir(folder)
+
+def folder_exists(client, folder):
+    """
+    Helper function to see if a directory exists in the sftp server
+    """
+    if folder not in client.listdir():
+        return False
+    return True
+
+
+def move_sftp_file(
+        client,
+        source_full_path,
+        destination_full_path):
+    """
+    Move file from SFTP server.
+    """
+    # check if file exists
+    try:
+        client.stat(source_full_path)
+    except FileNotFoundError:
+        sys.exit(ec.EXIT_CODE_INVALID_FILE_PATH)
```

This is failing silently. Needs to notify the user with a print statement about what happened.